lib/file-source/src/file_server.rs

```diff
@@ -23,6 +24,19 @@ use tokio::time::delay_for;
 
 use crate::paths_provider::PathsProvider;
 
+#[derive(Debug, Snafu)]
+pub enum FileServerError<
+    PathsSource: std::error::Error + 'static,
+    ChannelClosedSource: std::error::Error + 'static,
+> {
+    #[snafu(display("File paths could not be retrieved."))]
+    PathsError { source: PathsSource },
+    #[snafu(display("Output channel closed."))]
+    ChannelClosedError { source: ChannelClosedSource },
```

Comment:
Do these have more than one type for `source`? If not, they should just use that concrete type.

Reply:
https://github.com/timberio/vector/blob/05e7359b139a942fb285e932a996fa4fdf48c1c3/src/sources/file.rs#L240

and

https://github.com/timberio/vector/blob/05e7359b139a942fb285e932a996fa4fdf48c1c3/src/sources/kubernetes_logs/mod.rs#L297-L298

use the same type, used by the only callers to `FileServer::run` (as far as I can see).

However, since `C` in `FileServer::run` is generic, I don't think I can restrict `<C as Sink<Vec<(Bytes, String)>>>::Error` to a conrete type.

Reply:
`PathsSource` currently is either `Glob::Error` or `K8sPathsProvider::Error`.